app/lib/frontend/templates/layout.dart

```diff
@@ -103,40 +101,53 @@ String renderLayoutPage(
 
 String _renderSearchBanner({
   @required PageType type,
-  @required String platform,
-  @required String sdk,
   @required String publisherId,
   @required SearchQuery searchQuery,
 }) {
+  final sp = searchQuery != null
+      ? SearchPreference.fromSearchQuery(searchQuery)
+      : searchPreference ?? SearchPreference();
   final queryText = searchQuery?.query;
   final escapedSearchQuery =
       queryText == null ? null : htmlAttrEscape.convert(queryText);
+  bool includePreferencesAsHiddenFields = true;
```

Comment:
Could we change the logic, so this is off by default, and only included where needed. To avoid surprises of cases we did not consider.